src/firefly/js/core/LayoutCntlr.js

```diff
@@ -17,28 +23,27 @@ export const LO_MODE = new Enum(['expanded', 'standard']);
 
 /*---------------------------- Actions ----------------------------*/
 
+export const SET_LAYOUT         = `${LAYOUT_PATH}.setLayout`;
 export const SET_LAYOUT_MODE    = `${LAYOUT_PATH}.setLayoutMode`;
 export const SHOW_DROPDOWN      = `${LAYOUT_PATH}.showDropDown`;
-export const HIDE_DROPDOWN      = `${LAYOUT_PATH}.hideDropDown`;
 
 
 /*---------------------------- Reducers ----------------------------*/
 
-export function reducer(state={dropDown: {}}, action={}) {
-    const {mode, view} = action.payload || {};
+export function reducer(state={}, action={}) {
+    var {mode, view} = action.payload || {};
```

Comment:
This sounds good idea to have '|| {}'.  I guess this can also be applied to "rawAction"  usually passed in action creator functions which usually access property values from 'rawAction'.